src/librustc_typeck/check/method/probe.rs

```diff
@@ -128,7 +133,7 @@ pub enum PickKind<'tcx> {
                     ty::PolyTraitRef<'tcx>),
 }
 
-pub type PickResult<'tcx> = Result<Pick<'tcx>, MethodError<'tcx>>;
+pub type PickResult<'tcx> = Result<Vec<Pick<'tcx>>, MethodError<'tcx>>;
```

Comment:
Actually, this change was way easier. A few differences remain between when you look for a method name (which then should return only one method) and when you look for a returned type (and then, a lot can match). I might need to add another condition in the future with multiple possible matches and this seems to be the only correct solution without completely splitting the code.